File: template/springboot/TinyPro/src/main/java/com/TinyPro/entity/dto/UpdatePwdUserDto.java
package com.TinyPro.entity.dto;

import jakarta.validation.constraints.NotEmpty;
import lombok.Data;
import org.hibernate.validator.constraints.Length;

@Data
public class UpdatePwdUserDto {
    
    @NotEmpty(message = "{validation.email.notEmpty}")

    private String email;

    @NotEmpty(message = "{validation.token.notEmpty}")
    private String token;

    @NotEmpty(message = "{validation.newPassword.notEmpty}")
    @Length(min = 6, message = "{validation.newPassword.length}")
    private String newPassword;

    @NotEmpty(message = "{validation.oldPassword.notEmpty}")
    private String oldPassword;
}

File: template/nest-next/src/user/dto/update-pwd-user.dto.ts
import { PartialType } from '@nestjs/mapped-types';
import { CreateUserDto } from './create-user.dto';
import { IsNotEmpty } from 'class-validator';
import { i18nValidationMessage } from 'nestjs-i18n';
import { I18nTranslations } from '../../.generate/i18n.generated';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class UpdatePwdUserDto extends PartialType(CreateUserDto) {
  @ApiPropertyOptional({
    description: '电子邮箱'
  })
  email: string;
  @ApiPropertyOptional({
    description: 'AccessToken'
  })
  token: string;

  @ApiProperty({
    description: '新密码'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  newPassword: string;

  @ApiProperty({
    description: '老密码'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  oldPassword?: string;
}
